### PR TITLE
Add_mongo_ebs

### PR DESCRIPTION
**Pull request by jljalderman**
Add AWS EBS volume to mongo instances and reduce the root disk size.
This needs the corresponding change in govuk-puppet to create the mount etc.

### CODE REVIEW

**Comment by issyl0** [comment]
Looks like you need to line up some `=` in your Terraform to appease the linter.

**Comment by jljalderman** [comment]
done

File: terraform/projects/app-mongo/main.tf [hunk 1]
```diff
@@ -205,7 +243,26 @@ module "mongo-3" {
...
+
+resource "aws_volume_attachment" "external_volume_attachment" {
+  device_name = "/dev/sdf"
+  volume_id   = "${aws_ebs_volume.ebs-mongo-3.id}"
+  instance_id = "${aws_instance.mongo-3_eni.id}"
+}
+
+resource "aws_ebs_volume" "ebs-mongo-3" {
+  availability_zone = "${lookup(data.terraform_remote_state.infra_networking.private_subnet_names_azs_map, var.mongo_3_subnet)}"
+  type              = "gp2"
```
Comment (surminus) [inline terraform/projects/app-mongo/main.tf]: I wonder if it's worth considering `io1` type as we're mounting a database off it? Additional cost, but better performance.

**Comment by surminus** [comment]
The block to add the volumes seems different to the way we have mounted them previously in the code - we do it in a few other places ([example](https://github.com/alphagov/govuk-aws/blob/master/terraform/projects/app-logs-cdn/main.tf#L124-L149)). We create the mount with specific tags, ensure the instance is able to be able to query those tags, and then attach and mount the volume with [userdata](https://github.com/alphagov/govuk-aws/blob/master/terraform/userdata/10-attach-volumes). We ensure it is loaded in from adding it to the runtime [here](https://github.com/alphagov/govuk-aws-data/blob/master/data/app-logs-cdn/integration/common.tfvars#L3).

File: terraform/projects/app-mongo/main.tf (hunk 1 above)
Comment (issyl0) [reply terraform/projects/app-mongo/main.tf]: We have prior art for this in [app-graphite](https://github.com/alphagov/govuk-aws/blob/master/terraform/projects/app-graphite/main.tf#L209).

**Comment by jljalderman** [comment]
redoing

**Pull request closed without merging**